service/tutorial/order/OrderServices.xml

```diff
@@ -0,0 +1,338 @@
+                    <parameter name="orderparts" type="map">
+                        <parameter name="orderPartSeqId" required="true"/>
+                        <parameter name="partName" required="true"/>
+                        <parameter name="facilityId" required="true"/>
+                        <parameter name="shipmentMethodEnumId" required="true"/>
+                        <parameter name="partStatusId" required="true"/>
+                        <parameter name="partTotal" required="true"/>
+                        <parameter name="item_details" type="list">
+                            <parameter name="item_map" type="map">
+                                <parameter name="orderItemSeqId" required="true"/>
+                                <parameter name="productId" required="true"/>
+                                <parameter name="itemDescription" required="true"/>
+                                <parameter name="quantity" required="true"/>
+                                <parameter name="unitAmount" required="true"/>
+                            </parameter>
+                        </parameter>
+                    </parameter>
+                </parameter>
+            </parameter>
+        </out-parameters>
+        <actions>
+            <set field="customers_deatails" from="[:]"/>
+            <set field="orders" from="[]"/>
+            <set field="order_parts" from="[]"/>
+            <set field="item_details" from="[]"/>
+            <set field="persons" from="[]"/>
+            <!-- Getting all Order records of the OrderHeader entity -->
+            <entity-find entity-name="mantle.order.OrderHeader" list="orderList">
+                <select-field field-name="orderId,orderName,currencyUomId,placedDate,grandTotal"/>
+            </entity-find>
+            <!-- Iterate the orderList for getting order information -->
+            <iterate list="orderList" entry="orderMap">
+                <set field="orderId" from="orderMap.orderId"/>
+                <set field="orderName" from="orderMap.orderName"/>
+                <set field="currencyUomId" from="orderMap.currencyUomId"/>
+                <set field="salesChannelEnumId" from="orderMap.salesChannelEnumId"/>
+                <set field="statusId" from="orderMap.statusId"/>
+                <set field="placedDate" from="orderMap.placedDate"/>
+                <set field="grandTotal" from="orderMap.grandTotal"/>
+                <entity-find entity-name="mantle.order.OrderPart" list="orderPartList">
+                    <econdition field-name="orderId" from="orderMap.orderId"></econdition>
+                    <select-field
+                            field-name="orderId,orderPartSeqId,customerPartyId,partName,facilityId,shipmentMethodEnumId,statusId,partTotal"/>
+                </entity-find>
+                <iterate list="orderPartList" entry="orderPartMap">
+                    <entity-find-one entity-name="mantle.party.Person" value-field="personList">
```

Comment:
I think we can prepare the customer_details map outside the iterate.

Reply:
Yes, Janvi
We can also prepare the customer_details map outside the orderPartMap iterate.
I updated the recommended changes.
Thank you.